scheduler/util.go

```diff
@@ -376,11 +377,26 @@ func taintedNodes(state State, allocs []*structs.Allocation) (map[string]*struct
 	return out, nil
 }
 
-// shuffleNodes randomizes the slice order with the Fisher-Yates algorithm
-func shuffleNodes(nodes []*structs.Node) {
+// shuffleNodes randomizes the slice order with the Fisher-Yates
+// algorithm. We seed the random source with the eval ID (which is
+// random) to aid in postmortem debugging of specific evaluations and
+// state snapshots.
+func shuffleNodes(plan *structs.Plan, nodes []*structs.Node) {
+
+	// use the last 4 bytes because those are the random bits
+	// if we have sortable IDs
+	buf := []byte(plan.EvalID)
+	seed := int64(binary.BigEndian.Uint32(buf[len(buf)-4:]))
+
+	// include the plan result's RefreshIndex so that we don't retry with
+	// the exact same shuffle
+	seed |= planResult.RefreshIndex
```

Comment:
I think you want to use xor (`^`); plain or could result in a noop, e.g.
https://go.dev/play/p/NCrGoS4d0TK

Reply:
Seth wins the Bit Twiddling Award.